book/course/autotools.md

```diff
@@ -0,0 +1,470 @@
+checking arpa/inet.h presence... yes
+checking for arpa/inet.h... yes
+checking fcntl.h usability... yes
+checking fcntl.h presence... yes
+checking for fcntl.h... yes
+checking for inttypes.h... (cached) yes
+checking netdb.h usability... yes
+checking netdb.h presence... yes
+checking for netdb.h... yes
+checking stddef.h usability... yes
+checking stddef.h presence... yes
+checking for stddef.h... yes
+checking for stdint.h... (cached) yes
+checking for stdlib.h... (cached) yes
+checking for string.h... (cached) yes
+checking sys/socket.h usability... yes
+checking sys/socket.h presence... yes
+checking for sys/socket.h... yes
+checking for unistd.h... (cached) yes
+checking wchar.h usability... yes
+checking wchar.h presence... yes
+checking for wchar.h... yes
+checking for stdbool.h that conforms to C99... yes
+checking for _Bool... yes
+checking for inline... inline
+checking for int16_t... yes
+checking for int32_t... yes
+checking for int64_t... yes
+checking for int8_t... yes
+checking for off_t... yes
+checking for size_t... yes
+checking for uint16_t... yes
+checking for uint32_t... yes
+checking for uint64_t... yes
+checking for uint8_t... yes
+checking for _LARGEFILE_SOURCE value needed for large files... no
+checking for stdlib.h... (cached) yes
+checking for GNU libc compatible malloc... yes
+checking for stdlib.h... (cached) yes
+checking for GNU libc compatible realloc... yes
+checking for bzero... yes
+checking for gethostbyaddr... yes
+checking for gethostbyname... yes
+checking for memmove... yes
+checking for memset... yes
+checking for pow... no
+checking for select... yes
+checking for socket... yes
+checking for sqrt... no
+checking for strchr... yes
+checking for strdup... yes
+checking for strerror... yes
+checking for strstr... yes
+checking for strtol... yes
+checking for ftello... yes
+checking for library containing deflateSetHeader... -lz
+checking for library containing acosf... -lm
+checking for library containing pthread_create... -lpthread
+checking libunwind.h usability... no
+checking libunwind.h presence... no
+checking for libunwind.h... no
+checking for library containing unw_getcontext... no
+Using execinfo.h for backtrace.
+checking for .git/refs/heads/master... yes
+checking for .git/refs/heads/master... (cached) yes
+checking that generated files are newer than configure... done
+configure: creating ./config.status
+config.status: creating extern/samtools-1.3.1/htslib-1.3.1/Makefile
+config.status: creating extern/samtools-1.3.1/Makefile
+config.status: creating src/c/breseq/Makefile
+config.status: creating Makefile
+config.status: creating aux_build/config.h
+config.status: executing libtool commands
+config.status: executing depfiles commands
+```
+````
+
+You'll then got a lot of output showing it testing your system and ensuring it
+can find all the bits it needs.
+
+## Building
+
+Assuming everything's gone well so far, building the software is just a case of:
```

Comment:
Probably want a touch more detail here about what magic is happening. 

I.e. The configure step has created a top level Makefile that outlines the steps required for building the software and generating relevant executables/libraries. To execute these steps in the Makefile, we use the `make` command.